src/command.mli

```diff
@@ -0,0 +1,79 @@
+open! Stdune
+(** This module replaces arg_spec.mli in the post-arrow Dune. The main
+    change is the constructor [Dyn] which now takes [static t Build.s]
+    instead of ['a -> (unit, static) t], which allows us to get rid of
+    the first type parameter ['a] in [('a, 'b) Arg_spec.t]. *)
+
+(** Command line arguments specification *)
+
+(** This module implements a small DSL to specify the command line
+    argument of a program as well as the dependencies and targets of
+    the program at the same time.
+
+    For instance to represent the argument of [ocamlc -o src/foo.exe
+    src/foo.ml], one might write:
+
+    {[
+      [ A "-o"
+      ; Target (Path.relatie  dir "foo.exe")
+      ; Dep    (Path.relative dir "foo.ml")
+      ]
+    ]}
+
+    This DSL was inspired from the ocamlbuild API.  *)
+
+open! Import
+
+(** [A] stands for "atom", it is for command line arguments that are
+    neither dependencies nor targets.
+
+    [Path] is similar to [A] in the sense that it defines a command
+    line argument that is neither a dependency or target. However, the
+    difference between the two is that [A s] produces exactly the
+    argument [s], while [Path p] produces a string that depends on
+    where the command is executed. For instance [Path (Path.of_string
+    "src/foo.ml")] will translate to "../src/foo.ml" if the command is
+    started from the "test" directory.  *)
+
+type static = Static
+type dynamic = Dynamic
+
+type _ t =
```

Comment:
We should put this in a sub-module `Args`. `Command.t` makes me think of something that represent a command. But we can do this in a sub-sequence PR.

Reply:
Indeed. I like `Command.Args.t` better, but you know what name might be even better? `Arg_spec.t` :-)

Reply:
Happy to switch to `Command.Args.t` or back to `Arg_spec.t` :-) Let me know what you prefer.

Reply:
Done.